driver/ppm_fillers.c

```diff
@@ -5089,168 +5089,162 @@ int f_sys_open_by_handle_at_x(struct event_filler_arguments *args)
 	return add_sentinel(args);
 }
 
-int f_sys_io_uring_setup_x (struct event_filler_arguments *args)
+int f_sys_io_uring_setup_x(struct event_filler_arguments *args)
 {
-	int res;
-	unsigned long val;
-	unsigned long sq_entries = 0;
-	unsigned long cq_entries = 0;
-	unsigned long flags = 0;
-	unsigned long sq_thread_cpu = 0;
-	unsigned long sq_thread_idle = 0;
-	unsigned long features = 0;
-
-#ifdef __NR_io_uring_setup
-	struct io_uring_params params;
-#endif
-
-	int64_t retval = (int64_t)syscall_get_return_value(current, args->regs);
-	res = val_to_ring(args, retval, 0, false, 0);
-	if (unlikely(res != PPM_SUCCESS))
-		return res;
+	int res = 0;
+	long retval = 0;
+	unsigned long val = 0;
 
-	/* entries */
-	syscall_get_arguments_deprecated(current, args->regs, 0, 1, &val);
-	res = val_to_ring(args, val, 0, true, 0);
-	if (unlikely(res != PPM_SUCCESS))
-		return res;
+	/* All these params are sent equal to `0` if `__NR_io_uring_setup`
+	 * syscall is not defined.
+	 */
+	u32 sq_entries = 0;
+	u32 cq_entries = 0;
+	u32 flags = 0;
+	u32 sq_thread_cpu = 0;
+	u32 sq_thread_idle = 0;
+	u32 features = 0;
 
 #ifdef __NR_io_uring_setup
-	/*
-	 * io_uring_params: we get the data structure, and put its fields in the buffer one by one
-	 */
+	struct io_uring_params params = {0};
 	syscall_get_arguments_deprecated(current, args->regs, 1, 1, &val);
 	res = ppm_copy_from_user(&params, (void *)val, sizeof(struct io_uring_params));
-	if (unlikely(res != 0))
-		return PPM_FAILURE_INVALID_USER_MEMORY;
+	if(unlikely(res != 0))
+	{
+		sq_entries = 0;
```

Comment:
Don't we need to memset `params` to 0 here instead?

Reply:
yes, we can do that! ty!

Reply:
done ty!